### DISCUSSION TITLE
What is the purpose of `ChunkSender::tryGetPreviousChunk`

### DISCUSSION THREAD

**Discussion by gpalmer-latai** [eclipse-iceoryx/iceoryx#2226] [Q&A]
This function seems to be unused outside of unit tests. It appears to return a pointer to a chunk, which is NOT backed by reference counting and could be overwritten at any moment. Was this intended to be used at an earlier time for some poor man's publisher history?

**Comment by elfenpiff** [comment]
@gpalmer-latai I no longer remember. But if it's not used outside unit tests, I would say YAGNI.

**Comment by elBoberido** [comment]
@gpalmer-latai AFAIR this was once an optimization to reclaim the last sent chunk if it already was released by all subscriber. I think it was just forgotten to be removed from the lower layers.

Do you want to take care of the removal or shall I add it to my list?
